phasespace/fulldecay/fulldecay.py

```diff
@@ -0,0 +1,259 @@
+import itertools
+from typing import Callable, Union
+
+import tensorflow as tf
+import tensorflow.experimental.numpy as tnp
+from particle import Particle
+
+from phasespace import GenParticle
+
+from .mass_functions import _DEFAULT_CONVERTER
+
+_MASS_WIDTH_TOLERANCE = 0.01
+_DEFAULT_MASS_FUNC = "rel-BW"
+
+
+class FullDecay:
+    """A container that works like GenParticle that can handle multiple decays."""
+
+    def __init__(self, gen_particles: list[tuple[float, GenParticle]]):
+        """Create an instance of FullDecay.

```

Comment:
Rather move the docstring from the class down (that it creates an instance is self explanatory, but it usually shows this as *the* docstring. Or add something similar

Reply:
Fixed in 8627d134c5ec6935ce281b8c54f0e0c0cedeb54b